ngraph/test/backend/abs.in.cpp

```diff
@@ -32,35 +32,25 @@
 // clang-format on
 
 #include "gtest/gtest.h"
-#include "runtime/backend.hpp"
-#include "ngraph/runtime/tensor.hpp"
 #include "ngraph/ngraph.hpp"
-#include "util/all_close.hpp"
-#include "util/all_close_f.hpp"
-#include "util/ndarray.hpp"
+#include "util/engine/test_engines.hpp"
+#include "util/test_case.hpp"
 #include "util/test_control.hpp"
-#include "util/test_tools.hpp"
 
 using namespace std;
 using namespace ngraph;
 
 static string s_manifest = "${MANIFEST}";
+using TestEngine = test::ENGINE_CLASS_NAME(${BACKEND_NAME});
 
 NGRAPH_TEST(${BACKEND_NAME}, abs)
 {
     Shape shape{2, 2};
     auto A = make_shared<op::Parameter>(element::f32, shape);
     auto f = make_shared<Function>(make_shared<op::Abs>(A), ParameterVector{A});
 
-    auto backend = runtime::Backend::create("${BACKEND_NAME}");
-
-    // Create some tensors for input/output
-    auto a = backend->create_tensor(element::f32, shape);
-    copy_data(a, vector<float>{1, -2, 0, -4.75f});
-    auto result = backend->create_tensor(element::f32, shape);
-
-    auto handle = backend->compile(f);
-    handle->call_with_validate({result}, {a});
-    EXPECT_TRUE(test::all_close_f(
-        (vector<float>{1, 2, 0, 4.75f}), read_vector<float>(result), MIN_FLOAT_TOLERANCE_BITS));
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<float>({1, -2, 0, -4.75f});
+    test_case.add_expected_output<float>(shape, {1, 2, 0, 4.75f});
+    test_case.run();
```

Comment:
run() will trigger with default value of DEFAULT_FLOAT_TOLERANCE_BITS. Here MIN_FLOAT_TOLERANCE_BITS is required, and this value shall be passed together with run().

Reply:
@postrational is this MIN_FLOAT_TOLERANCE_BITS needed here for some reason ? Other tests use DEFAULT_FLOAT_TOLERANCE_BITS and I suppose that precision requirements should be the same in every test.

Reply:
In my case, omitting MIN_FLOAT_TOLERANCE_BITS into run causes the test to fail. By adding these flags present when calling test::all_close_f as in line 64-65 resolves the test fail.

Reply:
Tolerance is not the same for every test. Indeed, this is an important aspect of the test, otherwise it will fail on some backends, while it may pass on others.